types/types.go

```diff
@@ -23,67 +24,158 @@ import (
 	"github.com/prometheus/common/model"
 )
 
+type state uint8
+
+const (
+	Unprocessed state = iota
+	Active
+	Silenced
+	Inhibited
+)
+
+// allows tracking of the status of an alert (active, silenced, )
+type AlertStatus struct {
+	status state
+	value  []string
+}
+
+// custom serializer for alertStatus that handles providing default value
+// if alertStatus for given fingerprint is missing from the statusMap
+func (a *AlertStatus) MarshalJSON() ([]byte, error) {
+	status, found := statusMap[a.status]
+	if !found {
+		status = "unknown"
+	}
+
+	// TODO: Do we want to return all values?
```

Comment:
This would make sense for `Silenced` status as there might be multiple silences that match our alert. But this means that notify would need to check all, which I see that this PR is adding, so I would say yes, we should return everything.